detectors/cicd/gitlab/gitlab.go

```diff
@@ -0,0 +1,122 @@
+package gitlab
+
+import (
+	"context"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/sdk/resource"
+	semconv "go.opentelemetry.io/otel/semconv/v1.27.0"
+	"os"
+)
+
+const (
+	gitlabCIEnvVar           = "GITLAB_CI"
+	gitlabPipelineNameEnvVar = "CI_PIPELINE_NAME"
+	gitlabPipelineIdEnvVar   = "CI_PIPELINE_ID"
+	gitlabJobIdEnvVar        = "CI_JOB_ID"
+	gitlabJobNameEnvVar      = "CI_JOB_NAME"
+	gitlabJobStageEnvVar     = "CI_JOB_STAGE"
+	gitlabJobUrlEnvVar       = "CI_JOB_URL"
+
+	gitlabCommitRefNameEnvVar   = "CI_COMMIT_REF_NAME"
+	gitlabCommitTagEnvVar       = "CI_COMMIT_TAG"
+	gitlabMergeRequestIIDEnvVar = "CI_MERGE_REQUEST_IID"
+
+	gitlabProjectUrlEnvVar = "CI_PROJECT_URL"
+	gitlabProjectIDEnvVar  = "CI_PROJECT_ID"
+)
+
+type resourceDetector struct {
+}
+
+// compile time assertion that resourceDetector implements the resource.Detector interface.
+var _ resource.Detector = (*resourceDetector)(nil)
+
+// NewResourceDetector returns a [ResourceDetector] that will detect Gitlab Pipeline resources.
+func NewResourceDetector() resource.Detector {
+	return &resourceDetector{}
+}
+
+func (detector *resourceDetector) Detect(_ context.Context) (*resource.Resource, error) {
+
+	var attributes []attribute.KeyValue
+
+	isGitlabCI := os.Getenv(gitlabCIEnvVar) == "true"
+
+	if isGitlabCI {
+		attributes = append(attributes, detectCICDAttributes()...)
+		attributes = append(attributes, detectVCSAttributes()...)
+	}
+
+	return resource.NewWithAttributes(semconv.SchemaURL, attributes...), nil
+}
+
+// detectCICDAttributes https://github.com/open-telemetry/semantic-conventions/blob/main/docs/attributes-registry/cicd.md
+func detectCICDAttributes() []attribute.KeyValue {
+	var attributes []attribute.KeyValue
+
+	ciPipelineName := os.Getenv(gitlabPipelineNameEnvVar)
+	if ciPipelineName != "" {
+		attributes = append(attributes, attribute.String(string(semconv.CICDPipelineNameKey), ciPipelineName))
```

Comment:
You can adjust this kind of format. `semconv.CICDPipelineNameKey.String("xxxxx")`